web/src/compose.js

```diff
@@ -238,14 +267,23 @@ export function send_message(request = create_message_object()) {
             // (Restoring this state is handled by clear_compose_box
             // for locally echoed messages.)
             compose_ui.hide_compose_spinner();
-            return;
-        }
-
-        echo.message_send_error(message.id, response);
+        } else {
+            echo.message_send_error(message.id, response);
 
-        // We might not have updated the draft count because we assumed the
-        // message would send. Ensure that the displayed count is correct.
-        drafts.sync_count();
+            // We might not have updated the draft count because we assumed the
+            // message would send. Ensure that the displayed count is correct.
+            drafts.sync_count();
+        }
+        if (split_messages_enabled) {
+            if (rest_of_the_content) {
+                compose_state.message_content(
+                    content_to_send + compose_validate.SPLIT_DELIMITER + rest_of_the_content,
+                );
+            } else {
+                compose_state.message_content(content_to_send);
+            }
+            $("textarea#compose-textarea").trigger("input");
```

Comment:
Any reason for moving this line out here?

Reply:
Since the unsent part could be splittable, I am triggering the `input` event so that the split message banner appears. If removed, the user would have to type something for the banner to appear.

Reply:
Also, currently, if a message is restored from drafts, and can be split (and the option to split is enabled), no banner appears. I think we should trigger an input event when restoring content from drafts.

Reply:
I meant to ask, why did you move out this line here when you had initially put it in `compose_state.get_or_set`? That original change had made sense to me since it was more generic, was there any problem that you changed it?

Reply:
Oh that's because some of the tests were failing on putting that line in `get_or_set`.

Reply:
In cases like this, it should be the tests that are fixed to handle the change in code (provided the code is implementing correct behavior -- you can verify that on CZO), and not the code that is written around old tests. You can always ask in [#development help](https://chat.zulip.org/#narrow/stream/49-development-help) for help regarding puppeteer tests

Reply:
@kuv2707 is this still unresolved?

Reply:
It is resolved. [This comment](https://github.com/zulip/zulip/pull/28974#issuecomment-2045755730) explains what's wrong in triggering `input` event in `compose_state.get_or_set`.